Finjector.Web/ClientApp/src/components/Folders/DeleteFolder.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from "react";
+import { Modal, ModalHeader, ModalBody, ModalFooter } from "reactstrap";
+import { useDeleteFolderMutation } from "../../queries/folderQueries";
+import { useNavigate } from "react-router-dom";
+import FinjectorButton from "../Shared/FinjectorButton";
+
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faTrash } from "@fortawesome/free-solid-svg-icons";
+
+interface Props {
+  folderId: string;
+}
+
+const DeleteFolder = (props: Props) => {
+  const navigate = useNavigate();
+
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const deleteMutation = useDeleteFolderMutation();
+
+  const toggleModal = () => {
+    setModalOpen(!modalOpen);
+  };
+
+  const handleDelete = () => {
+    deleteMutation.mutate(props.folderId, {
+      onSuccess: () => {
+        navigate("/teams");
+        toggleModal();
+      },
+    });
+  };
+
+  return (
+    <>
+      <FinjectorButton onClick={toggleModal}>
+        <FontAwesomeIcon icon={faTrash} />
+        Delete Folder
+      </FinjectorButton>
+      <Modal isOpen={modalOpen} toggle={toggleModal}>
+        <ModalHeader tag="h2" toggle={toggleModal}>
+          Delete Folder
+        </ModalHeader>
+        <ModalBody>
+          Are you sure you want to delete this folder? This folder, any Chart
+          Strings within it will be removed.
+        </ModalBody>
+        <ModalFooter>
+          <FinjectorButton color="secondary" onClick={toggleModal}>
+            Cancel
+          </FinjectorButton>
+          <FinjectorButton
+            color="danger"
+            onClick={handleDelete}
+            disabled={deleteMutation.isLoading}
+          >
+            <FontAwesomeIcon icon={faTrash} />
+            Delete
```

Comment:
Optional but i like to change the text to "Deleting..." or some other loading state while doing the mutation

Reply:
Can do, but this was a direct copy from the DeleteTeam.tsx
Want me to change it there too?

Reply:
@srkirkland That is the dialog button text
![image](https://github.com/ucdavis/finjector/assets/469010/2188910b-67ba-434e-b75d-41829bcb73e7)

Reply:
Oh.. I see what you mean. Not sure how to do that. This was just a copy of the team delete.

Reply:
@jSylvestre I don't understand, i don't see that in the code anywhere

Reply:
@jSylvestre  { mutation.isLoading ? "Deleting" : "Delete" }

Reply:
I looked at all the code you had for delete team, then replicated it for delete folder.
Initially, I thought you meant change the text Delete to Deleting... (Didn't notice you said while the mutation was running.)